test/csuite/config/main.c

```diff
@@ -0,0 +1,494 @@
+/*-
```

Comment:
Can we have tests for overlapping keys in sub-configs and various types of incorrect inputs?

Also, it'd be great to add a test calling an API with an "unbound config".

Reply:
I added a test for unbound config (in test/csuite/config/main.c).

I'm not sure what you mean by "overlapping keys in sub-configs".  Eventually, when we support more APIs, and there are keys known by multiple APIs, we'll want to check that.  But right now, begin_transaction just has a single sub-config ("roundup_timestamps") with two possible keys ("prepared", "read"), and test/csuite/config tests valid and invalid values for each.

test/csuite/config/main.c tests lots of incorrect inputs - check_compiling_invalid_configurations() sets every key to multiple invalid values.  Is there something else you're thinking of?

Reply:
Re: `"cconf"` for compiled config.  There are 20 places that `WT_CONF *conf` declaration appears, and each has several uses.  I'm okay with changing them if you think it helps readability - just wanted to make sure that's what you're looking for.

Reply:
> Re: `"cconf"` for compiled config. There are 20 places that `WT_CONF *conf` declaration appears, and each has several uses. I'm okay with changing them if you think it helps readability - just wanted to make sure that's what you're looking for.

Thanks Don.
I was also thinking about function names. Plain "conf" looks to me like a contraction of the word "config" and it's not apparent how it's different. Something like "cconf" adds that special bit to it.

Reply:
Thanks, I'm going to push this suggestion to a followup ticket, in the interest of getting the baseline of work done.